app/services/id_token_builder.rb

```diff
@@ -55,24 +57,48 @@ def timestamp_claims
   end
 
   def acr
-    ial = identity.ial
-    case ial
-    when Idp::Constants::IAL_MAX then determine_ial_max_acr
-    when Idp::Constants::IAL1 then Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
-    when Idp::Constants::IAL2 then Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
-    else
-      raise "Unknown ial #{ial}"
+    return nil unless identity.acr_values.present?
+    acr_ial_component_values.map do |component_value|
+      if component_value == Vot::LegacyComponentValues::IALMAX
+        determine_ial_max_acr.name
+      else
+        component_value.name
+      end
+    end.join(' ')
+  end
+
+  def sp_requests_vot?
+    return false unless identity.vtr.present?
+    IdentityConfig.store.use_vot_in_sp_requests
+  end
+
+  def vot
+    return nil unless identity.vtr.present?
+    resolved_authn_context_result.component_values.map(&:name).join('.')
+  end
+
+  def acr_ial_component_values
+    resolved_authn_context_result.component_values.select do |component_value|
+      component_value.name.include?('ial')
```

Comment:
I believe that this could include an `loa` ACR value which may trip this up. You may want to consider something like this:

```ruby
def acr_value_for_ial
  if resolved_authn_context_result.identity_proofing?
    Vot::LegacyComponentValues::IAL2.name
  else
    Vot::LegacyComponentValues::IAL2.name
end
```

This approach will return the ial2 ACR value when identity proofing is requested (this is assuming it was done correctly which the checks in the authorization controller should ensure). When identity proofing is not requested this will return the ial1 ACR value.

Reply:
Actually I thought about this some and posted a shortcut here: https://github.com/18F/identity-idp/pull/10142#discussion_r1503079144